### PR TITLE
General mechanism to identify all possible GitHub urls

### LINKED ISSUE #618: Not all Github SCM url gets recognized
An example scm url from a pom.xml https://github.com/apache/rocketmq/blob/master/pom.xml#L38

```
<scm>
        <url>git@github.com:apache/rocketmq.git</url>
        <connection>scm:git:git@github.com:apache/rocketmq.git</connection>
        <developerConnection>scm:git:git@github.com:apache/rocketmq.git</developerConnection>
        <tag>HEAD</tag>
</scm>
```

Does not get recognized by rating-core.

DOD:
- Identify the possible syntaxes used in `<scm>` tag. And try to have a generalized approach to resolve a github project.
- Make sure that rating-core can identify such github syntaxes and identify them as a github repo.

### PR DESCRIPTION
**Pull request by sourabhsparkala**
Fixes #618

### CODE REVIEW

**Comment by sourabhsparkala** [comment]
The above fix is based on how the `<scm><url>` is defined in few projects mentioned below:
- https://repo1.maven.org/maven2/net/java/jvnet-parent/5/jvnet-parent-5.pom
- https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-databind/2.12.5/jackson-databind-2.12.5.pom
- https://repo1.maven.org/maven2/commons-io/commons-io/2.11.0/commons-io-2.11.0.pom
- https://repo1.maven.org/maven2/org/bouncycastle/bcprov-jdk15on/1.69/bcprov-jdk15on-1.69.pom
- https://repo1.maven.org/maven2/org/springframework/integration/spring-integration-core/5.5.3/spring-integration-core-5.5.3.pom
- https://repo1.maven.org/maven2/commons-io/commons-io/2.11.0/commons-io-2.11.0.pom

Also, the URI syntax was implemented based from here, which indicates what syntaxes are suitable to be used as URL specifications for Git
- https://maven.apache.org/scm/git.html
- https://git-scm.com/docs/git-clone#_git_urls

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/MavenScmFinder.java [hunk 1]
```diff
@@ -55,7 +58,55 @@
...
+  }
+ 
+  /**
+   * The method tries to resolve a GitHub project path from the given URI syntax.
+   * 
+   * @param url The input URL to be parsed and converted into a GitHub URL.
+   * @return A GitHub URL is parsing is successful. Otherwise an #Optional.empty().
+   * @throws IOException If something goes wrong.
+   */
+  static Optional<String> gitHubUrlParser(String url) throws IOException {
```
Comment (SebastianWolf-SAP) [inline src/main/java/com/sap/oss/phosphor/fosstars/tool/MavenScmFinder.java:71]: I probably wouldn't call this method `gitHubUrlParser`. IMHO methods should use verbs and also specify what they actually do. Maybe something like `resolveGitHubProjectPath` or `normalizeGitHubProjectPath` would be better and more understandable here.

**Review by SebastianWolf-SAP — CHANGES_REQUESTED** [review]
Hard to request changes just for such a small thingy, but sometimes I'm a nitpicker - just because code should be easy to grasp for people who are not that familiar with the details. Content-wise it seems to be fine and tests are also there, I would just recommend to change the method name to follow more the general code conventions: https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

File: src/main/java/com/sap/oss/phosphor/fosstars/tool/MavenScmFinder.java (hunk 1 above)
Comment (sourabhsparkala) [reply src/main/java/com/sap/oss/phosphor/fosstars/tool/MavenScmFinder.java:71]: I agree 👍

**Review by SebastianWolf-SAP — APPROVED** [review]

**Pull request merged**